### ISSUE TITLE
[BUG] Differences between GitHub comment and actual results

### ISSUE THREAD

**Issue by canihavesomecoffee** [CCExtractor/sample-platform#535]
Sample platform commit (found at the bottom of each page) : 

**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [x] I have checked that the issue I'm posting isn't already reported.
- [x] I have checked that the issue I'm posting isn't already solved and no duplicates exist in [closed issues](https://github.com/CCExtractor/sample-platform/issues?q=is%3Aissue+is%3Aclosed) and in [opened issues](https://github.com/CCExtractor/sample-platform/issues)
- [x] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

**My familiarity with the project is as follows (check one, eg [X]):**

- [ ] I have never visited/used the platform.
- [ ] I have used the platform just a couple of times.
- [ ] I have used the platform extensively, but have not contributed previously.
- [x] I am an active contributor to the platform.

---

The posted results to a PR on GitHub are inconsistent with what you see when you visit the actual results page.

For example:

https://github.com/CCExtractor/ccextractor/pull/1329#issuecomment-827744099 (table mentions fails, below it says all tests passed)
vs
https://sampleplatform.ccextractor.org/test/3224 (all tests passed)

Another example:
https://github.com/CCExtractor/ccextractor/pull/1329#issuecomment-827495365 (~19 listed failures)
vs
https://sampleplatform.ccextractor.org/test/3225 (5 fails)

**Comment by pranavrajpal** [comment]
I'd like to try working on this.

Is there a way to download the server's database so that I can try and look at the test results more closely? Right now, I can't really tell whether the correct test results are the ones shown on the Sample Platform website, the ones shown in the GitHub comment, or neither of them.

**Comment by kvshravan** [comment]
@pranavrajpal  I don't think you'll be needing the server database, sample database which comes through the installation must be enough, I suppose the issue is with the logic of [comment_pr function](https://github.com/CCExtractor/sample-platform/blob/master/mod_ci/controllers.py#L1148), especially while counting for the passed tests.

**Comment by canihavesomecoffee** [comment]
The correct results are on the page, so it's indeed the commenting function logic which goes wrong for some reason.

**Comment by kvshravan** [comment]
@pranavrajpal Are you still working on this? If you're stuck somewhere , we're ready to help :)

**Comment by pranavrajpal** [comment]
Yes, I'm still working on this.

Currently, I'm still trying to determine what the cause of this is. I _think_ that the problem is somehow related to regression tests that have multiple expected variants of a single output file, because several of the regression tests I've found that are incorrectly marked as failing in the comment in CCExtractor/ccextractor#1329 seem to have 2 variants for output file 1 (so far I've found [2](https://sampleplatform.ccextractor.org/regression/test/2/view), [19](https://sampleplatform.ccextractor.org/regression/test/19/view), and [104](https://sampleplatform.ccextractor.org/regression/test/104/view)). However, I'm still trying to reproduce this so I can't say for certain.

I have a question about how regression tests with multiple variants of the same file (like the previously mentioned tests) are stored in the database. The relevant model definitions for how the output files are stored seem to be `RegressionTestOutput`: https://github.com/CCExtractor/sample-platform/blob/e0cacb97af17c52b17a41225ee8315608e71e0ac/mod_regression/models.py#L136-L148
and `RegressionTestOutputFiles`:
https://github.com/CCExtractor/sample-platform/blob/e0cacb97af17c52b17a41225ee8315608e71e0ac/mod_regression/models.py#L213-L224

If there are multiple allowed variants of an output file, would that be represented as:
- a `RegressionTestOutput` with `correct` describing the original version of the file and `multiple_files` describing the variants?
- a `RegressionTestOutput` with `multiple_files` describing both the original version and the variants of the output files and `correct` being ignored?
- multiple `RegressionTestOutput` instances all pointing to the same regression test (either using `regression_test` or `regression_id`)?
- something else entirely?

**Comment by kvshravan** [comment]
Yes , you're correct. `RegressionTestOutput` model is used to store the original version of the output and `multiple_files` will fetch us the variant files. The testsuite checks if the output for a particular regression test, matches with the original output. If the output matches the `got` field will be set to `None`, else the `got` field will contain hash of the output file. A test is passed if `got` is `None` or if the hash value of `got` is one of the `multiple_files`. You can refer to [Pass/fail mechanism](https://github.com/CCExtractor/sample-platform/blob/master/mod_test/controllers.py#L134-L152) in mod_test/controllers.py. This should help you deal with the issue.

**Comment by devmrfitz** [comment]
@canihavesomecoffee Is this issue fixed or still open?

**Comment by thealphadollar** [comment]
Hey @devmrfitz , only part of the issue has been solved by #540 . Please refer the PR to know more :)
